src/pages/workspace/reimburse/WorkspaceRateAndUnitPage.tsx

```diff
@@ -0,0 +1,157 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useEffect} from 'react';
+import {Keyboard, View} from 'react-native';
+import FormProvider from '@components/Form/FormProvider';
+import InputWrapper from '@components/Form/InputWrapper';
+import type {OnyxFormValuesFields} from '@components/Form/types';
+import OfflineWithFeedback from '@components/OfflineWithFeedback';
+import Picker from '@components/Picker';
+import TextInput from '@components/TextInput';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as CurrencyUtils from '@libs/CurrencyUtils';
+import getPermittedDecimalSeparator from '@libs/getPermittedDecimalSeparator';
+import Navigation from '@libs/Navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@libs/Navigation/types';
+import * as NumberUtils from '@libs/NumberUtils';
+import * as PolicyUtils from '@libs/PolicyUtils';
+import withPolicy from '@pages/workspace/withPolicy';
+import type {WithPolicyProps} from '@pages/workspace/withPolicy';
+import WorkspacePageWithSections from '@pages/workspace/WorkspacePageWithSections';
+import * as BankAccounts from '@userActions/BankAccounts';
+import * as Policy from '@userActions/Policy';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import type {Unit} from '@src/types/onyx/Policy';
+
+type WorkspaceRateAndUnitPageProps = WithPolicyProps & StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.RATE_AND_UNIT>;
+
+type ValidationError = {rate?: string};
+
+function WorkspaceRateAndUnitPage({policy, route}: WorkspaceRateAndUnitPageProps) {
+    const {translate, toLocaleDigit} = useLocalize();
+    const styles = useThemeStyles();
+
+    useEffect(() => {
+        if ((policy?.customUnits ?? []).length !== 0) {
+            return;
+        }
+
+        BankAccounts.setReimbursementAccountLoading(true);
+        Policy.openWorkspaceReimburseView(policy?.id ?? '');
+    }, [policy?.customUnits, policy?.id]);
+
+    const unitItems = [
+        {label: translate('common.kilometers'), value: CONST.CUSTOM_UNITS.DISTANCE_UNIT_KILOMETERS},
+        {label: translate('common.miles'), value: CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES},
+    ];
+
+    const saveUnitAndRate = (unit: Unit, rate: string) => {
+        const distanceCustomUnit = Object.values(policy?.customUnits ?? {}).find((customUnit) => customUnit.name === CONST.CUSTOM_UNITS.NAME_DISTANCE);
+        if (!distanceCustomUnit) {
+            return;
+        }
+        const currentCustomUnitRate = Object.values(distanceCustomUnit?.rates ?? {}).find((r) => r.name === CONST.CUSTOM_UNITS.DEFAULT_RATE);
+        const unitID = distanceCustomUnit.customUnitID ?? '';
+        const unitName = distanceCustomUnit.name ?? '';
+        const rateNumValue = PolicyUtils.getNumericValue(rate, toLocaleDigit) as number;
```

Comment:
What if we use Number(rateNumValue) instead of assertion?
```suggestion
        const rateNumValue = PolicyUtils.getNumericValue(rate, toLocaleDigit);
```